src/custom/utils/trade.ts

```diff
@@ -99,14 +102,32 @@ export async function signAndPostOrder(params: PostOrderParams): Promise<AddUnse
 
   let signingScheme: SigningScheme
   let signature: string | undefined
-  if (allowsOffchainSigning) {
-    const signedOrderInfo = await signOrder(unsignedOrder, chainId, signer)
+  let orderStatus: OrderStatus
+
+  if (isSmartContractWallet) {
+    debugger
+    const signedOrderInfo = await signOrder(unsignedOrder, chainId, signer, isSmartContractWallet).catch((err) => {
```

Comment:
I would think that here, you have to attempt a EIP-1271, and if it fails you fallback to `presign`, rigtht?

Reply:
yes, if the signature is rejected with a Method not found(should be the standard), we fallback to the original plan